### PR TITLE
[Fix] Add inputs to qnn deserialize, add serialization of noise and mitigation in QM

### PR DESCRIPTION
**Pull request by dominikandreasseitz**
Fixes https://github.com/pasqal-io/qadence/issues/324

### CODE REVIEW

File: qadence/models/qnn.py [hunk 1]
```diff
@@ -218,3 +222,43 @@ def _format_to_dict(self, values: Tensor) -> dict[str, Tensor]:
...
+
+        qnn: QNN
+        try:
+            qm_dict = d[cls.__name__]
+            qnn = cls(
+                circuit=QuantumCircuit._from_dict(qm_dict["circuit"]),
+                observable=[deserialize(q_obs) for q_obs in qm_dict["observable"]],  # type: ignore[misc]
+                backend=qm_dict["backend"],
+                diff_mode=qm_dict["diff_mode"],
+                measurement=Measurements._from_dict(qm_dict["measurement"]),
```
Comment (Roland-djee) [inline qadence/models/qnn.py:248]: Don't you miss the `noise` parameter ? On that note, if QNN is a QM I guess the `mitigation` parameter should be present.

**Review by Roland-djee — COMMENTED** [review]
Thanks @dominikandreasseitz. I think there are parameters missing.

**Review by gvelikova — APPROVED** [review]

File: qadence/models/qnn.py (hunk 1 above)
Comment (jpmoutinho) [reply qadence/models/qnn.py:248]: It seems noise and mitigation are not there in the serialization of `QuantumModel` as well

Comment (dominikandreasseitz) [reply qadence/models/qnn.py:248]: added

**Pull request merged**